ZichanYang324/BF-Employee-HR
# BGP-19 [Backend]  visa status managment page

# Description
This PR implements enhancements to the document management system for users on OPT visas, focusing on the backend functionality to enforce a specific document upload order and enable HR to update the status of these documents. The changes ensure that users can only upload documents in a predefined sequence and receive appropriate feedback based on the document's status (Pending, Approved, Rejected) set by HR. This functionality is crucial for managing work authorization documents efficiently and ensuring compliance with the required document submission process.
Fixes # (bug is related which...)
Or Implements # (BGP) 

## Related PRS (if any):
1. Enhanced Document Upload Logic:
2. Improved Document Retrieval
3. Database Schema Adjustments
…

## How to test:
1. run `npm install` to ensure all dependencies are up to date.
2. Use 1npm run dev2` or the appropriate command to start the server locally.
3. Utilize Postman or a similar tool to test the document upload functionality by making a POST request to http://localhost:3100/documents/upload 
Test the document status update feature by making a PATCH request to http://localhost:3100/documents/:documentId/status, 
Retrieve the list of documents and their statuses by making a GET request to http://localhost:3100/documents/my 

## Screenshots or videos of changes or expecting result:


## Note:
Rightnow, I just store the document uploaded in MongoDB. There is a sperate auth middleware for this stage which will be updated later.
